### ISSUE TITLE
ESLint fails when trying to call functions named "component"

### ISSUE THREAD

**Issue by samuelthomas2774** [vuejs/eslint-plugin-vue#417]
**Tell us about your environment**

* **ESLint Version:** v4.16.0
* **eslint-plugin-vue Version:** 4.2.0
* **Node Version:** v6.9.1

**Please show your full configuration:**

```yml
---

parserOptions:
  ecmaVersion: 8
  sourceType: module

env:
  browser: true
  node: true

plugins:
  - vue

extends:
  - eslint:recommended
  - plugin:vue/recommended

rules:
  no-unused-vars: 'off'
  no-console: 'off'
  no-empty: 'off'
```

**What did you do? Please include the actual source code causing the issue.**

```js
component();
```

**What did you expect to happen?**

Nothing.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint outputs this and exits with status 1:

```
Cannot read property 'properties' of undefined
TypeError: Cannot read property 'properties' of undefined
    at Object.getComputedProperties (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint-plugin-vue/lib/utils/index.js:326:51)
    at Object.assign.utils.executeOnVue (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint-plugin-vue/lib/rules/no-async-in-computed-properties.js:133:42)
    at CallExpression:exit (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint-plugin-vue/lib/utils/index.js:472:9)
    at listeners.(anonymous function).forEach.listener (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (native)
    at Object.emit (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.leaveNode (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint/lib/util/node-event-generator.js:303:14)
    at CodePathAnalyzer.leaveNode (/Users/Samuel/Documents/BetterDiscord/2018-01-19/betterdiscordapp/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:630:23)
```

**Comment by samuelthomas2774** [comment]
[Fixed in version 4.3.0.](https://github.com/vuejs/eslint-plugin-vue/compare/v4.2.0...v4.3.0#diff-eae83c5838640e1c85538d8c9fbe35e3L391)
